### PR TITLE
Inverse

### PR DESCRIPTION
**Pull request by lesliechase**

### CODE REVIEW

**Comment by lesliechase** [comment]
I tried to test to see if the GTest passed but make wouldn't work because it didn't have DotProduct as a target? Which is weird because that was on a different branch.

**Comment by yiskylee** [comment]
I can take a look today. Will you be here at around 9?

**Comment by lesliechase** [comment]
Yes

**Comment by lesliechase** [comment]
Xiangyu I am still getting same error when testing my code. At first, I did not include Integer as a type and the tests failed in Google Test comparing each number (which were identical). When I included integer it didn't work. Does inverse return a new type? I committed my last changes so you can look at my latest Google Test file.

File: cpp/test/cpu_operations_test/Inverse_test.cc [hunk 1]
```diff
@@ -46,34 +47,36 @@ class InverseTest : public ::testing::Test {
 };
 
 
-typedef ::testing::Types<float, double> MyTypes;
+typedef ::testing::Types<int, float, double> MyTypes;
```
Comment (yiskylee) [inline cpp/test/cpu_operations_test/Inverse_test.cc]: if you take out int from MyTypes here, what were the errors?

**Comment by lesliechase** [comment]
The test file is adjusted and this should be all set to merge now.

File: cpp/test/cpu_operations_test/Inverse_test.cc [hunk 2]
```diff
@@ -47,7 +47,7 @@ class InverseTest : public ::testing::Test {
 };
 
 
-typedef ::testing::Types<int, float, double> MyTypes;
+typedef ::testing::Types</*int,*/float, double> MyTypes;
```
Comment (yiskylee) [inline cpp/test/cpu_operations_test/Inverse_test.cc]: /\* and */ are not recommended in google code style, could you just take out int?

File: cpp/test/cpu_operations_test/Inverse_test.cc [hunk 3]
```diff
@@ -62,7 +62,7 @@ TYPED_TEST(InverseTest, InverseFunctionality) {
   this->GetInverse();
   for (int i = 0; i < 3; i++) {
     for (int j = 0; j < 3; j++) {
-      EXPECT_EQ(this->output(i, j), this->correct(i, j));
+      ASSERT_NEAR(this->output(i, j), this->correct(i, j), 0.0001);
```
Comment (yiskylee) [inline cpp/test/cpu_operations_test/Inverse_test.cc]: Assert will cause the test exit for any assert failure, I suggest we use EXPECT_NEAR here.

File: cpp/test/cpu_operations_test/Inverse_test.cc (hunk 2 above)
Comment (lesliechase) [inline cpp/test/cpu_operations_test/Inverse_test.cc]: Interesting I had that taken out I must have commited my latest version but not pushed it to my repository to reflect the changes. Sorry!

**Comment by yiskylee** [comment]
@lesliechase , could you change the the name from Inverse_test.cc and DotProduct_test.cc to inverse_test.cc and dot_product_test.cc? And running "make check" shows that you have two errors at line 102 and 103 in cpu_operations.h, could you fix that as well? Please make sure to do a git pull upstarem develop between fixing anything.

**Pull request merged**